core/src/test/java/org/opensearch/sql/expression/datetime/DateTimeFunctionTest.java

```diff
@@ -1411,6 +1414,78 @@ public void secondOfMinuteInvalidArguments() {
     );
   }
 
+  private static Stream<Arguments> getTestDataForStrToDate() {
+    return Stream.of(
+        //Date arguments
+        Arguments.of(
+            "01,5,2013",
+            "%d,%m,%Y",
+            new ExprDateValue("2013-05-01"),
+            DATE),
+        Arguments.of(
+            "May 1, 2013",
+            "%M %d, %Y",
+            new ExprDateValue("2013-05-01"),
+            DATE),
+
+        Arguments.of(
+            "9,23,11",
+            "%h,%i,%s",
+            new ExprTimeValue("09:23:11"),
+            TIME),
+
+        Arguments.of(
+            "May 1, 2013 - 9,23,11",
+            "%M %d, %Y - %h,%i,%s",
+            new ExprDatetimeValue("2013-05-01 09:23:11"),
+            DATETIME),
+
+        //Invalid Arguments (should return null)
+        Arguments.of(
+            "a09:30:17",
+            "a%h:%i:%s",
+            ExprNullValue.of(),
+            UNDEFINED),
+        Arguments.of(
+            "abc",
+            "abc",
+            ExprNullValue.of(),
+            UNDEFINED),
+        Arguments.of(
+            "9",
+            "%m",
+            ExprNullValue.of(),
+            UNDEFINED),
+        Arguments.of(
+            "9",
+            "%s",
+            ExprNullValue.of(),
+            UNDEFINED)
+    );
+  }
+
+  @ParameterizedTest(name="{0} | {1}")
+  @MethodSource("getTestDataForStrToDate")
+  public void test_str_to_date(
+      String datetime,
+      String format,
+      ExprValue expectedResult,
+      ExprCoreType expectedType) {
+
+    lenient().when(nullRef.valueOf(env)).thenReturn(nullValue());
+    lenient().when(missingRef.valueOf(env)).thenReturn(missingValue());
+
+    FunctionExpression expression = DSL.str_to_date(
```

Comment:
Test with `date_format` and `time_format`. Something like:

```java
str_to_date(date_format(date, format), format) == date
```
and vice versa
```java
date_format(str_to_date(str, format), format) == str
```

Reply:
Fixed in https://github.com/Bit-Quill/opensearch-project-sql/pull/231/commits/0ab10018ddbb44e181e89883e8c69b16825daa05